### ISSUE TITLE
commonEnvironmentSettings being clobbered in job templates

### ISSUE THREAD

**Issue by dkreg** [Azure/azure-batch-cli-extensions#45]
The command `az batch job create --template job.json` appears to overwrite the value of `commonEnvironmentSettings` unconditionally with `null`. If the job is instead created using `az batch job create --json-file job.json`, then `commonEnvironmentSettings` is preserved.

**Comment by xingwu1** [comment]
Looking this issue.

**Comment by xingwu1** [comment]
Investigate the issue.

**Comment by xingwu1** [comment]
@dkreg, what do you mean "overwrite"? The pool does not exist before you call the command, so there no existing "commonEnvironmentSettings" to be preserved.

I tested that the template can set the commonEnvironmentSettings.

**Comment by dkreg** [comment]
@xingwu1 Thanks for looking into this. For one of my jobs, I found that, once the pool was created, the "commonEnvironmentSettings" property could be used to set environment variables for the job so long as the {"name": ..., "value": ...} pairs were given literal values. However, if the values were instead specified as template parameters, then they were automatically being replaced with 'null'.

I just tested this again, using a different json file, but was unable to recreate the problem - it seems to work now. Unfortunately, I no longer have the json files that originally exhibited the bug, so I can't replicate the exact conditions.
